src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/Failover/GetAzureStorSimpleFailoverVolumeContainers.cs

```
@@ -43,7 +42,14 @@ public override void ExecuteCmdlet()
                 switch(ParameterSetName)
                 {
                     case StorSimpleCmdletParameterSet.IdentifyById:
-                        deviceid = DeviceId;
+                        if (StorSimpleClient.IsValidDeviceId(DeviceId))
+                        {
+                            deviceid = DeviceId;
+                        }
+                        else
+                        {
+                            WriteVerbose(string.Format(Resources.NoDeviceFoundWithGivenIdInResourceMessage, StorSimpleContext.ResourceName, DeviceId));
```

We should throw an error.